006_buildkite_importing_pipelines/README.md

```diff
@@ -0,0 +1,374 @@
+
+:computer: `terminal`
+```sh
+mkdir -p internal/ci/buildkite
+```
+
+You may change the hierarchy and naming of `buildkite`'s **parent** directories
+to suit your repository layout. If you do so, you will need to adapt some
+commands and CUE code as you follow this guide.
+
+Move the newly-created CUE files into their dedicated directory. For example:
+
+:computer: `terminal`
+```sh
+mv ./.buildkite/*.cue internal/ci/buildkite
+```
+
+### Validate workflows
+
+<!-- TODO: upstream JSONSchema isn't usable
+cf. https://github.com/cue-lang/cue/issues/2698
+cf. https://github.com/cue-lang/cue/issues/2699
+
+#### :arrow_right: Fetch a pipeline schema
+
+Fetch a schema for Buildkite pipelines, as defined by Buildkite themselves, and
+place it in the `internal/ci/buildkite` directory:
+
+:computer: `terminal`
+```sh
+curl -o internal/ci/buildkite/buildkite.pipeline.schema.json https://raw.githubusercontent.com/buildkite/pipeline-schema/6396f68d5e983e0d2acbf829c565027a4cfd69bc/schema.json
+```
+
+We use a specific commit from the upstream repository to make sure that this
+process is reproducible.
+
+#### :arrow_right: Import the schema
+
+Import the schema into CUE:
+
+:computer: `terminal`
+```sh
+cue import -f -l '#Pipeline:' internal/ci/buildkite/buildkite.pipeline.schema.json
+```
+-->
+
+#### :arrow_right: Create a pipeline schema
+
+Create a very basic CUE schema for Buildkite pipelines, adapted from
+[Buildkite's documentation](https://buildkite.com/docs/pipelines/configuration-overview),
+and place it in the `internal/ci/buildkite` directory:
+
+:floppy_disk: `internal/ci/buildkite/buildkite.pipeline.schema.cue`
+
+```CUE
+package buildkite
+
+#Pipeline: {
+	steps!: [...]
+	env?: [string]: string
+	agents?: {[string]: string} | [ ..._#kv]
+	_#kv:    =~".+="
+	notify?: [...]
+}
+```
+
+|  :grey_exclamation: Info :grey_exclamation: |
+|:------------------------------------------- |
+| It would be great if we could use [Buildkite's authoritative pipeline schema](https://github.com/buildkite/pipeline-schema) here. Unfortunately, CUE's JSONSchema support can't currently import it. This is being tracked in CUE Issues [#2698](https://github.com/cue-lang/cue/issues/2698) and [#2699](https://github.com/cue-lang/cue/issues/2699), and this guide should be updated once the schema is useable.
+
+#### :arrow_right: Apply the schema
+
+We need to tell CUE to apply the schema to each pipeline.
+
+To do this we'll create a file at `internal/ci/buildkite/pipelines.cue` in our
+example.
+
+However, **if the pipeline imports that you performed earlier *already* created
+a file with that same path and name**, then simply select a different CUE
+filename that *doesn't* already exist. Place the file in the
+`internal/ci/buildkite/` directory.
+
+:floppy_disk: `internal/ci/buildkite/pipelines.cue`
+
+```
+package buildkite
+
+// each member of the pipelines struct must be a valid #Pipeline
+pipelines: [_]: #Pipeline
+```
+
+### Generate YAML from CUE
```

Comment:
I would add a sentence of context here to be explicit about the fact that we are going to generate YAML that will ultimately replace the source files from which we originally imported.

Reply:
Done.